lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java

```diff
@@ -1565,6 +1542,73 @@ private void updateTaskPrioDurationFromAttachments(TaskImpl task, PrioDurationHo
         return result;
     }
 
+    private void deleteTask(String taskId, boolean forceDelete)
+        throws TaskNotFoundException, InvalidStateException, NotAuthorizedException {
+        LOGGER.debug("entry to deleteTask(taskId = {} , forceDelete = {})", taskId, forceDelete);
+        taskanaEngine.checkRoleMembership(TaskanaRole.ADMIN);
+        TaskImpl task = null;
+        try {
+            taskanaEngine.openConnection();
+            task = (TaskImpl) getTask(taskId);
+
+            if (!TaskState.COMPLETED.equals(task.getState()) && !forceDelete) {
+                throw new InvalidStateException("Cannot delete Task " + taskId + " because it is not completed.");
+            }
+            taskMapper.delete(taskId);
+            LOGGER.debug("Task {} deleted.", taskId);
+        } finally {
+            taskanaEngine.returnConnection();
+            LOGGER.debug("exit from deleteTask().");
+        }
+    }
+
+    void addTaskTransferredEvent(Task task, String previousWorkbasketId, String newWorkbasketId) {
+        if (HistoryEventProducer.isHistoryEnabled()) {
+            historyEventProducer.createEvent(new TaskTransferredEvent(task, previousWorkbasketId, newWorkbasketId));
+        }
+    }
+
+    void addTasksTransferredEvent(List<MinimalTaskSummary> taskSummaries, TaskSummaryImpl updateObject) {
```

Comment:
And this one would probably better named createTaskTransferredEvents, don't you think?